packages/material-ui-styles/src/withStyles/withStyles.js

```diff
@@ -67,6 +68,25 @@ const withStyles = (stylesOrCreator, options = {}) => (Component) => {
       if (withTheme && !more.theme) {
         more.theme = theme;
       }
+
+      if (theme && theme.variants && theme.variants[name]) {
+        let variantsClasses = '';
+        const themeVariants = theme.variants[name];
+
+        themeVariants.forEach((themeVariant) => {
+          let isMatch = true;
+          Object.keys(themeVariant.props).forEach((key) => {
+            if (more[key] !== themeVariant.props[key]) {
+              isMatch = false;
+            }
+          });
+          if (isMatch) {
+            variantsClasses = `${variantsClasses} ${classes[propsToClassKey(themeVariant.props)]}`;
+          }
+        });
+
+        classes = { ...classes, root: `${classes.root}${variantsClasses}` };
```

Comment:
@oliviertassinari why the blank space was removed? Isn't this going to append the classes wrongly?
```js
classes.root = "a b c"
variantsClasses = "d e"
result = "a b cd e"
```
I believe this is wrong and we need to have black space between the two as it was before.

Reply:
I have noticed leading spaces in the classes object. From what I understand the existing append logic force one ` ` hence the case considered can't be reached.

Reply:
Shouldn't we rather remove the leading space from `variantsClasses` then? The code can easily be misinterpreted to create a single class in `root`

Reply:
This is not an issue now that the logic is separated in a hook and the classes are actually merged by `clsx`